kmath-tensors/src/commonMain/kotlin/space/kscience/kmath/tensors/core/algebras/DoubleLinearOpsTensorAlgebra.kt

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2018-2021 KMath contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package space.kscience.kmath.tensors.core.algebras
+
+import space.kscience.kmath.tensors.api.LinearOpsTensorAlgebra
+import space.kscience.kmath.nd.as1D
+import space.kscience.kmath.nd.as2D
+import space.kscience.kmath.tensors.api.TensorStructure
+import space.kscience.kmath.tensors.core.*
+import space.kscience.kmath.tensors.core.checkSquareMatrix
+import space.kscience.kmath.tensors.core.choleskyHelper
+import space.kscience.kmath.tensors.core.cleanSymHelper
+import space.kscience.kmath.tensors.core.luHelper
+import space.kscience.kmath.tensors.core.luMatrixDet
+import space.kscience.kmath.tensors.core.luMatrixInv
+import space.kscience.kmath.tensors.core.luPivotHelper
+import space.kscience.kmath.tensors.core.pivInit
+import kotlin.math.min
+
+
+public class DoubleLinearOpsTensorAlgebra :
+    LinearOpsTensorAlgebra<Double>,
+    DoubleTensorAlgebra() {
+
+    override fun TensorStructure<Double>.inv(): DoubleTensor = invLU(1e-9)
+
+    override fun TensorStructure<Double>.det(): DoubleTensor = detLU(1e-9)
+
+    public fun TensorStructure<Double>.luFactor(epsilon: Double): Pair<DoubleTensor, IntTensor> =
+        computeLU(tensor, epsilon) ?:
+        throw RuntimeException("Tensor contains matrices which are singular at precision $epsilon")
+
+    public fun TensorStructure<Double>.luFactor(): Pair<DoubleTensor, IntTensor> = luFactor(1e-9)
+
```

Comment:
Missing documentation, especially about the components of the resulting tuple (what are L, U, and P in it?).